torch/testing/_internal/common_utils.py

```diff
@@ -292,20 +338,39 @@ def clear_tracked_input():
 # Wraps an iterator and tracks the most recent value the iterator produces
 # for debugging purposes. Tracked values are stored on the test function.
 class TrackedInputIter:
-    def __init__(self, child_iter, input_type_desc, callback=lambda x: x):
+    def __init__(self, child_iter, input_type_desc,
+                 callback=lambda x: x, set_seed=True, restrict_to_index=None):
         self.child_iter = enumerate(child_iter)
         # Input type describes the things we're tracking (e.g. "sample input", "error input").
         self.input_type_desc = input_type_desc
         # Callback is run on each iterated thing to get the thing to track.
         self.callback = callback
         self.test_fn = extract_test_fn()
+        # Indicates whether the random seed should be set before each call to the iterator
+        self.set_seed = set_seed
+        # Indicates that iteration should be restricted to only the provided index.
+        # If None, no restriction is done
+        self.restrict_to_index = restrict_to_index
 
     def __iter__(self):
         return self
 
     def __next__(self):
-        # allow StopIteration to bubble up
-        input_idx, input_val = next(self.child_iter)
+        while True:
+            if self.set_seed:
+                # use a test-name-specific hash for the seed if possible
+                seed = (
+                    int.from_bytes(hashlib.sha256(
+                        self.test_fn.__qualname__.encode("utf-8")).digest()[:4], 'little')
+                    if self.test_fn is not None else SEED
+                )
+                set_rng_seed(seed)
+
+            # allow StopIteration to bubble up
+            input_idx, input_val = next(self.child_iter)
+            if (self.restrict_to_index is None) or (input_idx == self.restrict_to_index):
```

Comment:
ahhh this conditional took me a bit. I was confused why you'd have the first `if self.restrict_to_index is None` and it's because you wanna feed it forward! Cuz if it's specified, you want to ignore every other value and only feed the indexed one forward. Smart.

Reply:
Is there a way in the nonempty restrict to index case to tell the iterator to short circuit to the end once the index has been found? vs needing to keep going next?

Reply:
> Is there a way in the nonempty restrict to index case to tell the iterator to short circuit to the end once the index has been found? vs needing to keep going next?

let's see, we could manually raise `StopIteration` once we've passed the restricted index. This approach won't exhaust the child iterator, but maybe that's okay?